office_space_allocation/amity.py

```diff
@@ -0,0 +1,150 @@
+import random
+
+from office_space_allocation.fellow import Fellow
+from office_space_allocation.office import Office
+from office_space_allocation.person import Person
+from office_space_allocation.staff import Staff
+from office_space_allocation.utilities import InvalidRoomOccupantError, RoomFullError, MultiplePeopleFoundException
+
+
+class Amity:
+    def __init__(self):
+        self.all_persons = []
+        self.all_rooms = []
+        self.allocated_rooms = []
+
+    def add_person(self, person):
+        """
+        Adds a new Person to list of persons
+        :param: ```Person``` person
+        """
+        if not isinstance(person, Person):
+            raise TypeError("Argument should be of type Person")
+        else:
+            self.all_persons.append(person)
+
+    def add_room(self, new_room):
+        """
+        Adds ```Room``` new_room to list of all_rooms
+        :param ```Room``` new_room
+        """
+        self.all_rooms.append(new_room)
+
+    def has_room(self, room):
+        """
+        Checks if Room `room` already exists in Amity
+        :return: `True` if room exist, otherwise `False`
+        """
+        for r in self.all_rooms:
+            if r == room:
+                return True
+        return False
+
+    def find_room(self, name):
+        """
+        Find a ```Room``` object using name
+        :param name: ```str``` name of the Room to look for
+        :return: ```Room``` object if found
+        """
+        name = name.lower()
+        for r in self.all_rooms:
+            if name == r.get_name().lower():
+                return r
+        raise ValueError(name + " not found!")
+
+    def allocate_room(self, person):
+        """
+        Allocates a room to ```Person``` person.
+
+        This is done randomly.
+        :param person: ```Person``` to allocate room
+        :return: room : ```Room``` allocated , if possible
+        """
+        if isinstance(person, Staff):
+            # allocate only offices
+            offices = [of for of in self.all_rooms if isinstance(of, Office)]
+            if len(offices) < 1:
+                raise IndexError("There are no more Office rooms to allocate. Please create some rooms before making "
+                                 "allocations")
+            else:
+                rm = random.choice(offices)
+
+                rm.add_person(person)
+                self.allocated_rooms.append(rm)
+                return rm
+        elif isinstance(person, Fellow):
+            # raise exception if there are no rooms
+            if len(self.all_rooms) < 1:
+                raise IndexError("There are no available rooms. Please create some rooms before making allocations.")
+            else:
+                office_rm = random.choice(self.all_rooms)
+
+                office_rm.add_person(person)
```

Comment:
Yeah that's right. I have actually fixed that bug. Please check the updated code in the develop branch of this repo.